processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -347,14 +347,23 @@ public static SegmentAnalysis mergeAnalyses(
       mergedId = "merged";
     }
 
+    final Boolean rollup;
+
+    if (arg1.isRollup() != null && arg2.isRollup() != null && arg1.isRollup().equals(arg2.isRollup())) {
+      rollup = arg1.isRollup();
+    } else {
+      rollup = null;
```

Comment:
there is a case, if previous segment is rolled up, and the next segment use new spec which is not to rollup, then for segmentMetadataQuery should handle this case and return null if merge=true
Otherwise, rollup should be always true or false